demisto_sdk/commands/common/hook_validations/modeling_rule.py

```diff
@@ -50,6 +70,77 @@ def is_schema_file_exists(self):
                 return False
         return True
 
+    @error_codes("MR104")
+    def is_schema_types_valid(self):
+        """
+            Validates all types used in the schema file are valid, i.e. part of the list below.
+        """
+        valid_types = {'string', 'int', 'float', 'datetime', 'boolean'}
+        invalid_types = []
+        if self.schema_content:
+            for dataset in self.schema_content:
+                attributes = self.schema_content.get(dataset)
+                for attr in attributes.values():
+                    type_to_validate = attr.get('type')
+                    if type_to_validate not in valid_types:
+                        invalid_types.append(type_to_validate)
+
+            if invalid_types:
+                error_message, error_code = Errors.modeling_rule_schema_types_invalid(invalid_types)
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    self._is_valid = False
+                    return False
+        return True
+
+    @error_codes("MR105")
+    def dataset_name_matches_in_xif_and_schema(self):
+        """
+            Validates the dataset name is the same in the xif file and in the schema file
+        """
+
+        def get_dataset_from_xif(xif_file_path: str) -> List[str]:
+            with open(xif_file_path, 'r') as xif_file:
+                xif_content = xif_file.read()
+                dataset = re.findall("dataset[ ]?=[ ]?([\"a-zA-Z_0-9]+)", xif_content)
+                if dataset:
+                    return [dataset_name.strip("\"") for dataset_name in dataset]
```

Comment:
please unindent these three lines, they don't need to be under the `with` statement